### PR TITLE
Add example to erase NVM

### PR DESCRIPTION
**Pull request by sandeepmistry**
Adding this based on the discussion in https://github.com/ArmDeveloperEcosystem/lorawan-library-for-pico/issues/14#issuecomment-1017049643, https://github.com/ArmDeveloperEcosystem/lorawan-library-for-pico/issues/14#issuecomment-1017164162, and https://github.com/ArmDeveloperEcosystem/lorawan-library-for-pico/issues/14#issuecomment-1018400307.

### CODE REVIEW

File: examples/erase_nvm/main.c [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+int main( void )
+{
+    // initialize stdio and wait for USB CDC connect
+    stdio_init_all();
+
+    while (!tud_cdc_connected()) {
+        tight_loop_contents();
+    }
+
```
Comment (AiyionPrime) [inline examples/erase_nvm/main.c]:
```suggestion

```

File: examples/erase_nvm/main.c [hunk 2]
```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) 2021 Arm Limited and Contributors. All rights reserved.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ * 
+ *
```
Comment (AiyionPrime) [inline examples/erase_nvm/main.c]:
```suggestion
 *
 *
```

File: examples/erase_nvm/main.c [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        .miso = PICO_DEFAULT_SPI_RX_PIN,
+        .sck  = PICO_DEFAULT_SPI_SCK_PIN,
+        .nss = 8
+    },
+    .reset = 9,
+    .dio0 = 7,
+    .dio1 = 10
+};
+
+// LoRaWAN region to use, full list of regions can be found at:
```
Comment (AiyionPrime) [inline examples/erase_nvm/main.c]:
```suggestion
// LoRaWAN region to use, full list of regions can be found at:
```

**Review by AiyionPrime — COMMENTED** [review]
Looks good, compiles, and reports to be successful via tty.
There are several trailing spaces, you might want to get rid of.

**Comment by ThomasLeister** [comment]
I was already thinking of submitting similar changes ;-) Thanks! This should definitively help new users of the library.

**Comment by sandeepmistry** [comment]
@AiyionPrime @ThomasLeister thanks for the feedback!

@AiyionPrime I've pulled in your changes regarding trailing whitespace and rebased.

**Pull request merged**